test/Makefile

```diff
@@ -0,0 +1,22 @@
+tests = $(wildcard **/*.robot)
+RF_VENV :=../_output/.venv
+RF_BINARY :=../_output/.venv/bin/robot
+RF_OUT :=--outputdir ../_output/e2e-$$(date +''%Y%m%d-%H%M%S)
+
+.PHONY: setup dry-run with-filter with-tag all $(tests)
+
+setup:
+	python3 -m venv $(RF_VENV)  && \
+	$(RF_VENV)/bin/python3 -m pip install -r requirements.txt
+
+$(tests):
+	$(RF_BINARY) \
+		$(RF_OPTIONS) \
+		$(RF_OUT) \
+		$@
+
+dry-run: RF_OPTIONS=--dryrun
+dry-run: $(tests)
+
+all: RF_OPTIONS=-v USHIFT_HOST:$${USHIFT_HOST} -v USHIFT_USER:microshift -x xunit.xml
```

Comment:
Can we move all global variables to a variablefile in yaml format. This will allow the config file to be generated before starting the suite and not require exporting shell environment variables.

all: RF_OPTIONS=-x xunit.xml -L DEBUG --variablefile test_config.yaml

Example:
$ cat test/test_config.yaml 
USHIFT_HOST:    dhcp-1-235-103.arm.eng.rdu2.redhat.com
USHIFT_USER:    redhat
SSH_PRIV_KEY:   /home/microshift/.ssh/id_rsa